src/state/signalr-connection.ts

```diff
@@ -0,0 +1,139 @@
+            throw new Error('VUE_APP_SIGNALR_URL not defined, please add it to the environment');
+        }
+    }
+
+    public async subscribeEdition(editionId: number) {
+        console.debug('SignalR subscribeEdition called, current status ', this._status);
+        if (this._subscribedEditionId === editionId) {
+            return;
+        }
+
+        if (this._status !== 'connected') {
+            await this.connect();
+        }
+
+        if (this._subscribedEditionId) {
+            await this.unsubscribeEdition();
+        }
+
+        await this._utils!.subscribeToEdition(editionId);
+        console.debug('SubscribeToEdition called');
+        this._subscribedEditionId = editionId;
+    }
+
+    public async unsubscribeEdition() {
+        console.debug('SignalR unsubscribeEdition, status is ', this._status);
+        if (this._status !== 'connected' || !this._subscribedEditionId) {
+            return;
+        }
+
+        await this._utils!.unsubscribeToEdition(this._subscribedEditionId);
+        console.debug('UnsubscribeToEdition called');
+        this._subscribedEditionId = undefined;
+    }
+
+    public async userChanged() {
+        console.debug('SignalR changing users');
+        const subscribed = this._subscribedEditionId;
+        await this.disconnect();
+        await this.connect();
+        if (subscribed) {
+            await this.subscribeEdition(subscribed);
+        }
+    }
+
+    public registerNotificationHandler(handler: NotificationHandler) {
+        this.unregisterNotificationHandler();
+        this._currentHandler = handler;
+        if (this._utils) {
+            this._utils!.connectNotificationHandler(handler);
+        }
+    }
+
+    public unregisterNotificationHandler() {
+        if (this._currentHandler && this._utils) {
+            this._utils.disconnectNotificationHandler(this._currentHandler);
+        }
+        this._currentHandler = undefined;
+    }
+
+    private async connect() {
+        console.debug('SignalR connect called, current status ', this._status);
+        if (this._connection) {
+            this._connection.stop();
+        }
+
+        this._connection = new HubConnectionBuilder()
+            .withUrl(process.env.VUE_APP_SIGNALR_URL!, {
+                accessTokenFactory: () => StateManager.instance.session.token || ''
+            }).configureLogging(process.env.NODE_ENV === 'development' ? LogLevel.Information : LogLevel.Error)
+            .build();
+        this._utils = new SignalRUtilities(this._connection);
+
+        this._status = 'connecting';
+        try {
+            await this._connection.start();
+            this._status = 'connected';
+            this._connection!.onclose(this.onConnectionClosed);
+
+            if (this._currentHandler) {
+                this._utils.connectNotificationHandler(this._currentHandler);
+            }
+            console.debug('SignalR connection opened, status is ', this._status);
```

Comment:
probably implement autoreconnect and add your onReconnecting and onReconnected functions here.